amethyst_core/src/named.rs

```diff
@@ -1,6 +1,160 @@
-use std::borrow::Cow;
+use std::{borrow::Cow, ops::Deref};
 
-use specs::{world::LazyBuilder, Component, DenseVecStorage, EntityBuilder, WriteStorage};
+use fnv::FnvHashMap as HashMap;
+use specs::{
+    storage::MaskedStorage, world::LazyBuilder, Component, DenseVecStorage, Entity, EntityBuilder,
+    Storage, WriteStorage,
+};
+
+use util::{Cache, CachedStorage};
+
+/// Extension trait to find an entity based on its `Named` component.
+/// Can only be used on the `Named` component storage.
+pub trait FindNamed {
+    /// Finds an entity by its name by querying it from a map.
+    ///
+    /// For performance reasons, this should only be used once, and the returned `Entity` shall
+    /// be stored to avoid future lookups.
+    ///
+    /// # Conflict behavior
+    ///
+    /// Only the last `Named` component with name `s` will be returned; if
```

Comment:
"last" is very arbitrary and poorly defined from a user perspective, so we should probably make this an error case.  If a named entity that is expected to be unique is not unique that is extremely likely to be a coding error, and should be caught.

Reply:
I should clarify: I want this to return `Result`.

Reply:
I would actually prefer this to return all matched entities so the caller can decide what to do with them. Consider a scenario where you have a "Goblin" name in a prefab, and then spawn 300 goblins using that prefab, you'll have 300 entities with Named("Goblin"). If I wanted to apply some logic to all 300, retrieving all of them in this way could be beneficial.

Reply:
If you want you could add separate functions for each scenario, that use a base one that gives `Iterator<Item=Entity>` or some such.